### PR TITLE
Fix doctest CI

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix doctest CI, caused by a change in #20757. The change in multi-label example has some minor issue. See comments in the review.

### CODE REVIEW

File: src/transformers/utils/doc.py [hunk 1]
```diff
@@ -264,7 +264,9 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
     ...     "{checkpoint}", num_labels=num_labels, problem_type="multi_label_classification"
     ... )
 
-    >>> labels = torch.nn.functional.one_hot(torch.tensor(predicted_class_ids), num_classes=num_labels).to(torch.float)
+    >>> labels = torch.sum(
+    ...     torch.nn.functional.one_hot(predicted_class_ids[None, :].clone(), num_classes=num_labels), dim=1
+    ... ).to(torch.float)
```
Comment (ydshieh) [inline src/transformers/utils/doc.py:269]: Regarding `[None, :]` here, a few lines above, there is 
```
predicted_class_ids = torch.arange(0, logits.shape[-1])[torch.sigmoid(logits).squeeze() > 0.5]
```
(which is changed in #20757), which drops the batch dimension. When creating labels, we still need the batch dim, so

Comment (ydshieh) [reply src/transformers/utils/doc.py:269]: In current `main`, let's say `num_labels=6`, and `predicted_class_ids` has 2 elements (i.e. no batch dimension).

`torch.nn.functional.one_hot` will produce a tensor of `[2, 6]`, and each row is a one-hot tensor. However, what we want is a single tensor (i.e. only batch size 1) with 1s and 0s, not necessary one-hot. We have to sum the tensor along the selected class ids.

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

File: src/transformers/utils/doc.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/utils/doc.py:269]: FYI: need to add `dim=0` in `torch.sigmoid(logits).squeeze(dim=0)` for edge case where `logits` have shape `(1,1)`. This happens with `num_labels=1`, for example the checkpoint used for `GPT2ForSequenceClassification`. This doesn't really make sense for `multi-labels`, but I just make it work anyway.
